app/server/models/credit_transfer.py

```diff
@@ -341,4 +341,5 @@ def __init__(self,
         self.append_organisation_if_required(self.recipient_transfer_account.organisation)
         self.append_organisation_if_required(self.sender_transfer_account.organisation)
 
-        self.check_sender_transfer_limits()
+        # todo: [PR] this feels like backwards logic, in order to check limits, you need to init a credit transfer, which raises error. should only be on resolve_as_completed()
```

Comment:
Yeah can definitely see how checks make sense to be done outside of initialisation.
Have u made sure moving this doesn't break any of the USSD check balance functions?

Reply:
Yes, i've updated